pkg/webhook/loadbalancer/mutator.go

```diff
@@ -0,0 +1,132 @@
+)
+
+type mutator struct {
+	admission.DefaultMutator
+
+	namespaceCache ctlcorev1.NamespaceCache
+	projectCache   ctlmgmtcattlev3.ProjectCache
+	vmiCache       ctlkubevirtv1.VirtualMachineInstanceCache
+}
+
+var _ admission.Mutator = &mutator{}
+
+func NewMutator(namespaceCache ctlcorev1.NamespaceCache, projectCache ctlmgmtcattlev3.ProjectCache,
+	vmiCache ctlkubevirtv1.VirtualMachineInstanceCache) admission.Mutator {
+	return &mutator{
+		namespaceCache: namespaceCache,
+		projectCache:   projectCache,
+		vmiCache:       vmiCache,
+	}
+}
+
+func (m *mutator) Create(_ *admission.Request, newObj runtime.Object) (admission.Patch, error) {
+	lb := newObj.(*lbv1.LoadBalancer)
+
+	project, err := m.findProject(lb.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	var network string
+	if lb.Spec.WorkloadType == lbv1.Cluster && lb.Annotations != nil && lb.Annotations[utils.AnnotationKeyCluster] != "" {
+		network, err = m.findNetwork(lb.Namespace, lb.Annotations[utils.AnnotationKeyCluster])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	annotations := lb.Annotations
+	if len(annotations) == 0 {
+		annotations = map[string]string{}
+	}
+	annotations[utils.AnnotationKeyNamespace] = lb.Namespace
+	annotations[utils.AnnotationKeyProject] = project
+	annotations[utils.AnnotationKeyNetwork] = network
+
+	return []admission.PatchOp{
+		{
+			Op:    admission.PatchOpReplace,
+			Path:  "/metadata/annotations",
+			Value: annotations,
+		},
+	}, nil
+}
+
+func (m *mutator) Resource() admission.Resource {
+	return admission.Resource{
+		Names:      []string{"loadbalancers"},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   lbv1.SchemeGroupVersion.Group,
+		APIVersion: lbv1.SchemeGroupVersion.Version,
+		ObjectType: &lbv1.LoadBalancer{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+		},
+	}
+}
+
+// Find Project through the namespace
+func (m *mutator) findProject(namespace string) (string, error) {
+	ns, err := m.namespaceCache.Get(namespace)
+	if err != nil {
+		return "", fmt.Errorf("get namespace %s failed, error: %w", namespace, err)
+	}
+
+	projectID, ok := ns.Annotations[rancherproject.ProjectIDAnn]
+	if !ok {
+		return "", nil
+	}
+
+	fields := strings.Split(projectID, ":")
+	if len(fields) != 2 {
+		return "", fmt.Errorf("invalid project id %s", projectID)
+	}
+
+	project, err := m.projectCache.Get(fields[0], fields[1])
+	if err != nil {
+		return "", fmt.Errorf("get project %s/%s failed, error: %w", fields[0], fields[1], err)
+	}
+
+	return project.Spec.DisplayName, nil
+}
+
+// Find the first network where the guest cluster is running
+// We assume that all the virtual machines composed the guest cluster are running in the same namespace and
+// have the same network configuration.
+func (m *mutator) findNetwork(namespace, clusterName string) (string, error) {
+	// list all the vmi instance in the same namespace of the load balancer
+	vmis, err := m.vmiCache.List(namespace, labels.Everything())
```

Comment:
we can only list vmis created by node driver by using label `harvesterhci.io/creator=docker-machine-driver-harvester`